Revert `shootDNS` field removal (#7332)
gardener/gardener: docs/development/changing-the-api.md

#### Removing a field

> If fields shall be removed permanently from the API then a proper deprecation period must be adhered to so that end-users have enough time to adapt their clients.

Reviewer: ```suggestion
If fields shall be removed permanently from the API, then a proper deprecation period must be adhered to so that end-users have enough time to adapt their clients.
```

> Once the deprecation period is over field should be dropped from API in a two-step process i.e. it should be dropped in two release cycles. In the first step, all the usage from the code base should be dropped and In the second step, the field should be dropped from API. We need to follow this two-step process cause there can be the case where `gardener-apisever` is upgraded to a new version in which the field has been removed but other controllers are still on the old version of the gardener, this will lead to `nil` pointer exception.

Reviewer: ```suggestion
Once the deprecation period is over, the field should be dropped from the API in a two-step process, i.e., in two release cycles. In the first step, all the usages in the code base should be dropped. In the second step, the field should be dropped from API. We need to follow this two-step process cause there can be the case where `gardener-apiserver` is upgraded to a new version in which the field has been removed but other controllers are still on the old version of Gardener. This can lead to `nil` pointer exceptions or other unexpected behaviour.
```

> The followed process for removing a field is:

Reviewer: ```suggestion
The steps for removing a field from the code base is:
```
